api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -108,6 +108,13 @@ type HeaderSettings struct {
 	// is encountered. The default action is to reject the request.
 	// +optional
 	WithUnderscoresAction *WithUnderscoresAction `json:"withUnderscoresAction,omitempty"`
+
+	// PreserveExternalRequestID configures Envoy to keep the X-Request-ID header if passed for a request that is edge
+	// (Edge request is the request from external clients to front Envoy) and not reset it, which is the current Envoy behaviour.
+	// It defaults to false.
+	//
+	// +optional
+	PreserveExternalRequestID *bool `json:"preserveExternalRequestID,omitempty"`
```

Comment:
thoughts on `PreserveXRequestID` ?

the comments highlight the fact that this is opt in and needed when client is external / envoy is edge
cc @envoyproxy/gateway-maintainers

Reply:
actually, my initial plan was to use that naming, but I'm trying to mimic Envoy's naming conventions. i have changed just now to `PreserveXRequestID`